vmdb/db/migrate/20150619153304_miq_report_result_fix_serialized_report.rb

```diff
@@ -0,0 +1,53 @@
+class MiqReportResultFixSerializedReport < ActiveRecord::Migration
+  class MiqReportResult < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+
+    YAML_ATTRS = [:table, :sub_table, :filter_summary, :extras, :ids, :scoped_association, :html_title, :file_name, :extras, :record_id,
+                  :tl_times, :user_categories, :trend_data, :performance, :include_for_find, :report_run_time, :chart, :reserved]
+
+    def serialize_report_to_hash
+      val = read_attribute(:report)
+
+      if val.include?("!ruby/object:MiqReport")
+        val.sub!(/MiqReport/, 'Hash')
+      else
+        raise "unexpected format of report attribute encountered, '#{val.inspect}'"
+      end
+      raw_hash = YAML.load(val).last
+      new_hash = YAML_ATTRS.each_with_object(raw_hash['attributes'].to_hash) { |k, h| h[k.to_s] = raw_hash[k.to_s] }
+
+      YAML.dump(new_hash)
+    end
+
+    def serialize_hash_to_report
+      val = read_attribute(:report)
+
+      if val.starts_with?("---")
+        new_hash = {'attributes' => {}}
+        YAML.load(val).each do |k,v|
+          YAML_ATTRS.include?(k.to_sym) ? new_hash[k.to_s] = v : new_hash['attributes'][k.to_s] = v
+        end
+
+        YAML.dump([new_hash]).sub(/---\n- attributes:/, "---\n- !ruby/object:MiqReport\n  attributes:")
```

Comment:
Not sure if there an easier or more elegant way to do this but this is doing the job.